src/app/WriteClient.cpp

```diff
@@ -244,12 +244,21 @@ CHIP_ERROR WriteClient::SendWriteRequest(SessionHandle session, System::Clock::T
     // Create a new exchange context.
     mpExchangeCtx = mpExchangeMgr->NewContext(session, this);
     VerifyOrExit(mpExchangeCtx != nullptr, err = CHIP_ERROR_NO_MEMORY);
-    mpExchangeCtx->SetResponseTimeout(timeout);
+    if (session.IsGroupSession())
```

Comment:
Why should this conditional enabling of the `kExpectResponse` flag happen here? The ExchangeContext should automatically do this since responses on groupcast messages _on the same_ exchange aren't permitted.

Reply:
Because this function is used both for Unicast (with the kExpectResponse) and Groupcast (with no response expected)

Reply:
I the suggestion is that exchanges on a group session should ignore kExpectResponse (and probably ignore SetResponseTimeout calls) so we do that check in one place (in the exchange), not at all the exchange consumers.  That might make sense to do....

Reply:
I was gonna go one step further - that upon transmission, the EC would get closed by the exchange itself without further protocol/application involvement.

Reply:
That happens automatically if you send a message without `kExpectResponse`, yes.

Reply:
Issue #11631 was created to handle this comment